geoserver/geoserver_seeder/geoserver_seeder.rb

```diff
@@ -357,87 +49,74 @@ def seed_layers(layers, opts)
 #   - true on success, false otherwise
 #
 def geoserver_seeder_main(opts)
-  layers = []
 
-  if opts[:geonetwork]
-    # Get layers from Geonetwork URL
-    geonetwork_connector = GeonetworkConnector.new(opts[:geonetwork], opts[:geoserver])
-    layers = geonetwork_connector.wms_layers
-  else
-    # Get layers from command line
-    layers = opts[:layers]
-  end
+    geonetwork_connector = GeonetworkConnector.new(opts)
+
+    # layers that match supplied WMS server or all layers on Geonetwork
+    geonetwork_layers = geonetwork_connector.wms_layers
+    layers_to_seed = []
+
+    # add only layers from command line
+    if opts[:layers]
+        opts[:layers].each do | layer|
+            if(geonetwork_layers.any? {|geolayer| geolayer[1] == layer})
+                layers_to_seed.push([opts[:geoserver], layer])
```

Comment:
@julian1 The script would installed on whichever machine required so it can be called directly. Example usage is in geoserver_seeder.rb
```
    Seed Geoserver tiles based on results returned from Geonetwork

    Examples:
        (dry run) Seed Geowebcache with all layers in catalogue-imos.aodn.org.au, zoom level 2 to default:
           geoserver_seeder.rb -d -u https://catalogue-imos.aodn.org.au/geonetwork -s 2 

        (dry run) Seed  all layers in catalogue-imos.aodn.org.au from http://maps.aims.gov.au/geoserver/wms
            geoserver_seeder.rb -d -u https://catalogue-imos.aodn.org.au/geonetwork -g http://maps.aims.gov.au/geoserver/wms

        (dry run) Seed the argo and aatams layers, zoom level 4 to 6:
           geoserver_seeder.rb -T reseed  -d -u https://catalogue-portal.aodn.org.au/geonetwork -l imos:argo_profile_layer_map imos:aatams_sattag_nrt_profile_map -g http://geoserver-123.aodn.org.au/geoserver/wms  -s 4 -e 6 
```